File: app/src/main/java/com/niccher/mpesa_analyzer/interfaces/JsonProcesses.java
package com.niccher.mpesa_analyzer.interfaces;


import com.niccher.mpesa_analyzer.helpers.SummaryResponse;
import com.niccher.mpesa_analyzer.models.Mod_Loot_Summary;
import com.niccher.mpesa_analyzer.models.Mod_My_Loot_Count;

import java.util.Map;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.FieldMap;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

public interface JsonProcesses {

    @FormUrlEncoded
    @POST("get/my_uploads")
    Call <SummaryResponse> getSummary(@FieldMap Map<String, String> fields);

    @FormUrlEncoded
    @POST("get/my_summary_calculations")
    Call <Mod_Loot_Summary> getSummaryCalc(@FieldMap Map<String, String> fields);

    @FormUrlEncoded
    @POST("get/my_uploads_count")
    Call <Mod_My_Loot_Count> getLootCount(@FieldMap Map<String, String> fields);

    /*
    @FormUrlEncoded
    @POST("device")
    Call<Mod_Fone_Id> createPrint(@FieldMap Map<String, String> fields);
     */
}

File: app/src/main/java/com/niccher/my_mpesa_analyzer/interfaces/JsonProcesses.kt
package com.niccher.my_mpesa_analyzer.interfaces

import com.niccher.my_mpesa_analyzer.helpers.SummaryResponse
import com.niccher.my_mpesa_analyzer.models.Mod_My_Loot_Count
import retrofit2.Call
import retrofit2.http.FieldMap
import retrofit2.http.FormUrlEncoded
import retrofit2.http.POST

interface JsonProcesses {

    @FormUrlEncoded
    @POST("get/my_uploads")
    fun getSummary(@FieldMap fields: Map<String, String>): Call<SummaryResponse>

//    @FormUrlEncoded
//    @POST("get/my_summary_calculations")
//    fun getSummaryCalc(@FieldMap fields: Map<String, String>): Call<ModLootSummary>
//
    @FormUrlEncoded
    @POST("get/my_uploads_count")
    fun getLootCount(@FieldMap fields: Map<String, String>): Call<Mod_My_Loot_Count>
//
//    @FormUrlEncoded
//    @POST("get/my_uploads_category_count")
//    fun getLootCountCategories(@FieldMap fields: Map<String, String>): Call<SummaryLootResponse>
//
//    @FormUrlEncoded
//    @POST("get/list_all_sms_in_category")
//    fun getAllSmsInCategory(@FieldMap fields: Map<String, String>): Call<SummaryLootResponse>
//
//    @FormUrlEncoded
//    @POST("set/delete_loot_by_uuid")
//    fun getLootDelete(@FieldMap fields: Map<String, String>): Call<ModLootDelete>

}